### ISSUE TITLE
Issues in MLP and Attention Implementations

### ISSUE THREAD

**Issue by yyjiao97** [zhaoyiran924/Safety-Neuron#2]
1.For the MLP computation, the current implementation uses:
torch.sum(torch.abs(self.up_proj(x)), dim=1).squeeze().tolist()
This seems inconsistent with the algorithm described in the paper, as it omits the W_down component.
Based on my understanding, it should be:
torch.sum(((self.act_fn(self.gate_proj(x)) * self.up_proj(x)).unsqueeze(-1) * self.down_proj.weight.T)**2, dim=(1, -1)).squeeze().tolist()

2.For the attention mechanism, the attn_weights_temp (which corresponds to ∆(x) in the paper) does not seem to be normalized by √d as described in the paper.
